plugins/outputs/stackdriver/stackdriver.go

```diff
@@ -241,78 +244,83 @@ func (s *Stackdriver) Write(metrics []telegraf.Metric) error {
 }
 
 func getStackdriverTimeInterval(
-	m metricpb.MetricDescriptor_MetricKind,
-	start int64,
-	end int64,
-) (*monitoringpb.TimeInterval, error) {
-	switch m {
-	case metricpb.MetricDescriptor_GAUGE:
-		return &monitoringpb.TimeInterval{
-			EndTime: &timestamppb.Timestamp{
-				Seconds: end,
-			},
+	kind metric.MetricDescriptor_MetricKind,
+	value *monpb.TypedValue,
+	m telegraf.Metric,
+	f *telegraf.Field,
+	cc *CounterCache,
```

Comment:
I think most of those variables are required to get the`start` and `end`-time which was passed in directly before. So how about moving determination of start and end-time to an own function and leave this function as is?

Reply:
👍